### PR TITLE
Use npm as package manager for Javascript

### LINKED ISSUE #814: Minify web viewer resources
I think it would considerably speed up accessing the webserver, if the deployed style and script resources are minified in the deploy.

However, it should be toggle-able, as minification makes debugging a local deploy a lot harder.

### PR DESCRIPTION
**Pull request by csordasmarton**
* Use npm as package manager
* Minify JavaScript and CSS resources (closes #814)
* Build package and release targets

### CODE REVIEW

File: scripts/build_package.py [hunk 1]
```diff
@@ -493,6 +267,18 @@ def build_package(repository_root, build_package_config, env=None):
...
+    npm_modules = os.path.join(repository_root, 'vendor', 'node_modules')
+    yui_compress_jar = None
+
+    minify = build_package_config.get('minify')
+    if minify:
+        yui_compress_jar = os.path.join(repository_root, 'vendor',
+                                        'yuicompressor.jar')
+
+        if not os.path.exists(yui_compress_jar):
+            LOG.warn("YUI compressor can not be found at location %s",
```
[the remark this answers is not in the document]
Comment (gyorb) [reply scripts/build_package.py]: Please use `warning`.

File: scripts/build_package.py [hunk 2]
```diff
@@ -554,70 +342,52 @@ def build_package(repository_root, build_package_config, env=None):
...
-    file_url = dojo_dep['source_package']['url']
-    url_data = urlparse.urlparse(file_url)
-    _, file_name = os.path.split(url_data.path)
-    head, _ = file_name.split('.tar.gz')
-
-    dojo_root = os.path.join(repository_root, dojo_dep.get('directory'))
-    dojo_root = os.path.join(dojo_root, head)
-    target = os.path.join(package_root, package_layout['web_client_dojo'])
-    copy_tree(dojo_root, target)
+    if minify:
```
Comment (gyorb) [inline scripts/build_package.py]: A comment would be nice, that each js file which should be minified should be listed here.

File: vendor/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "CodeChecker",
+  "version": "1.0.0",
```
Comment (gyorb) [inline vendor/package.json]: Shouldn't we use the release version here?

File: scripts/build_package.py [hunk 4]
```diff
@@ -117,7 +111,23 @@ def create_folder_layout(path, layout):
                     sys.exit()
 
 
-def copy_tree(src, dst, skip=None):
+def yui_compress(source, destination, yui_compress_jar):
```
Comment (gyorb) [inline scripts/build_package.py]: Please add function documentation.

**Comment by csordasmarton** [comment]
With the new UI we use npm as a package manager so this pull request can be closed.

**Pull request closed without merging**